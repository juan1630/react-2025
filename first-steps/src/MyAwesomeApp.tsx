import type { CSSProperties } from "react";

const firstName = "Jose Juan";
const lastName = "Patrón";
const favoritesGames = ["Assasinss creed", "Call of duty", "Smash"];
const isActive = true;

const address = {
  zipcode: "abc-123",
  country: "México",
};

const myStyles: CSSProperties = {
  backgroundColor: "red",
  color: "white",
  padding: "10px",
  borderRadius: "10px",
  fontSize: "20px",
  textAlign: "center",
  fontWeight: "bold",
};
export function MyAwesomeApp() {
  return (
    <>
      <div data-testid="div-app">
        <h1 data-testid="first-name-title"> {firstName} </h1>
        <h3 className="main-title"> {lastName} </h3>
        <p> {favoritesGames.splice(1, 2)} </p>
        {2 + 2}
        <h1> {isActive ? "Activo" : "No está activo"} </h1>
        <p style={myStyles}> {JSON.stringify(address)} </p>
      </div>
    </>
  );
}
