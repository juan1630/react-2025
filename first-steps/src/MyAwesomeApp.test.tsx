import { describe, expect, test } from "vitest";
import { render, screen } from "@testing-library/react";
import { MyAwesomeApp } from "./MyAwesomeApp";

describe("MyAwesomeApp.tsx", () => {
  test("should render firstName and lastName", () => {
    const { container } = render(<MyAwesomeApp />);

    screen.debug();
    const h1 = container.querySelector("h1");
    expect(h1?.innerHTML).toContain("Jose Juan");

    const h3 = container.querySelector("h3");
    expect(h3?.innerHTML).toContain("PatrÃ³n");
  });

  test("should render firstName and lastName", () => {
  
    //El container solo cuando es un render inicial
    render(<MyAwesomeApp />);

    screen.debug();
    // const h1 = screen.getByRole('heading', { level: 1});
    const h1 = screen.getByTestId('first-name-title');
    console.log(h1.innerHTML)
    expect(h1?.innerHTML).toContain("Jose Juan");

  });

  test('should match snapshot', ()=> {
    const { container } = render(<MyAwesomeApp/>)
    expect(container).toMatchSnapshot()
  })

   test('should match snapshot', ()=> {
    
    render(<MyAwesomeApp/>)
    expect(screen.getByTestId('div-app')).toMatchSnapshot()
  })
});
