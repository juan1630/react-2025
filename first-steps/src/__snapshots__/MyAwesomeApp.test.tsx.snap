// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyAwesomeApp.tsx > should match snapshot 1`] = `
<div>
  <div
    data-testid="div-app"
  >
    <h1
      data-testid="first-name-title"
    >
       
      Jose Juan
       
    </h1>
    <h3
      class="main-title"
    >
       
      Patrón
       
    </h3>
    <p>
       
       
    </p>
    4
    <h1>
       
      Activo
       
    </h1>
    <p
      style="background-color: red; color: white; padding: 10px; border-radius: 10px; font-size: 20px; text-align: center; font-weight: bold;"
    >
       
      {"zipcode":"abc-123","country":"México"}
       
    </p>
  </div>
</div>
`;

exports[`MyAwesomeApp.tsx > should match snapshot 2`] = `
<div
  data-testid="div-app"
>
  <h1
    data-testid="first-name-title"
  >
     
    Jose Juan
     
  </h1>
  <h3
    class="main-title"
  >
     
    Patrón
     
  </h3>
  <p>
     
     
  </p>
  4
  <h1>
     
    Activo
     
  </h1>
  <p
    style="background-color: red; color: white; padding: 10px; border-radius: 10px; font-size: 20px; text-align: center; font-weight: bold;"
  >
     
    {"zipcode":"abc-123","country":"México"}
     
  </p>
</div>
`;
